scripts/gen_connector_charm.py

```diff
@@ -0,0 +1,910 @@
+        k: v for k, v in options.items() if k not in mandatory and k not in mandatory_with_default
+    }
+    sorted_options = {}
+    for k, v in sorted(mandatory.items(), key=operator.itemgetter(0)):
+        sorted_options[k] = v
+    for k, v in sorted(mandatory_with_default.items(), key=operator.itemgetter(0)):
+        sorted_options[k] = v
+    for k, v in sorted(optional.items(), key=operator.itemgetter(0)):
+        sorted_options[k] = v
+    return sorted_options
+
+
+def render_template(
+    *,
+    name,
+    connector_type,
+    version,
+    display_name,
+    config,
+    output_dir,
+    connector_name: str | None = None,
+    display_name_short: str | None = None,
+    charm_override: str = "",
+    generate_entrypoint: str = "",
+    install_location: str | None = None,
+    template_dir: pathlib.Path = pathlib.Path("connector-template"),
+):
+    if "_" in name or name.lower() != name:
+        raise ValueError(f"connector name should be in kebab case: {name}")
+    connector_name = connector_name or name
+    display_name_short = display_name_short or display_name
+    output_dir.mkdir(exist_ok=True)
+    for source in template_dir.glob("**/*"):
+        file = source.relative_to(template_dir)
+        if "__pycache__" in str(file):
+            continue
+        output = output_dir / file
+        if source.is_dir():
+            output.mkdir(exist_ok=True)
+            continue
+        if not str(file).endswith(".j2"):
+            output.write_bytes(source.read_bytes())
+            continue
+        output = pathlib.Path(str(output).removesuffix(".j2"))
+        template = jinja2.Template(source.read_text(), keep_trailing_newline=True)
+        template.globals["kebab_to_pascal"] = kebab_to_pascal
+        template.globals["constant_to_kebab"] = constant_to_kebab
+        output.write_text(
+            template.render(
+                name=name,
+                connector_name=connector_name,
+                connector_type=connector_type,
+                version=version,
+                display_name=display_name,
+                display_name_short=(
+                    display_name if display_name_short is None else display_name_short
+                ),
+                config=yaml.safe_dump(
+                    {"config": {"options": sort_config(config)}}, width=99999, sort_keys=False
+                ),
+                charm_override=charm_override,
+                install_location=(
+                    install_location if install_location else f"opencti-connector-{connector_name}"
+                ),
+                generate_entrypoint=generate_entrypoint,
+            ),
+            encoding="utf-8",
+        )
+    (output_dir / "lib/charms/opencti/v0").mkdir(parents=True, exist_ok=True)
+    (output_dir / "lib/charms/opencti/v0/opencti_connector.py").write_bytes(
+        pathlib.Path("lib/charms/opencti/v0/opencti_connector.py").read_bytes()
+    )
+    (output_dir / "src/charm.py").chmod(0o755)
```

Comment:
Could we please have some new lines to separate logical blocks of code

Reply:
Added, thanks!